File: programming_by_contract/java/ProgrammingByContract/src/main/java/corejava/Time.java
package corejava;

public class Time {
    private int hour;
    private int minute;

    public Time(int hour, int minute) {
        assert hour >= 0 && hour < 24 :
                "Hour out of range";
        assert minute >= 0 && minute < 60 :
                "Minute out of range";
        this.hour = hour;
        this.minute = minute;
    }


}

File: programming_by_contract/csharp/ProgrammingByContract/ProgrammingByContract/ProgrammingByContract/CodeContracts/Time.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.Contracts;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProgrammingByContract.CodeContracts
{
    public class Time
    {
        private int hour;
        private int minute;

        public Time(int hour, int minute)
        {
            Contract.Requires<ArgumentException>
                (hour >= 0 && hour <= 23);
            Contract.Requires(minute >= 0 && minute <= 59);
            this.hour = hour;
            this.minute = minute;
        }
    }
}
